lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -247,6 +247,20 @@ def tunit_edges(self, vals=None, M=None):
                  (tc[7], tc[4])]
         return edges
 
+    def apply_aspect(self, position=None):
+        if position is None:
+            position = self.get_position(original=True)
+
+        # in the superclass, we would go through and actually deal with axis
+        # scales and box/datalim. Those are all irrelevant - all we need to do
+        # is make sure our coordinate system is square.
+        figW, figH = self.get_figure().get_size_inches()
```

Comment:
Because all sizes in matplotlib are relative to the figure. `pb.shrunk_to_aspect` seems to take the ratio of the top-level parent (the figure), and the desired ratio.

Note that this is the same as the normal aspect=equal code, so is presumably correct.